spec/execution_tests.lua

```diff
@@ -0,0 +1,2163 @@
+local util = require "pallene.util"
+
+local execution_tests = {}
+
+function execution_tests.run(compile, backend, describe, it, assert)
+    local assert_test_output = function (expected)
+        local output = assert(util.get_file_contents("__test__output__.txt"))
+        assert.are.same(expected, output)
+    end
+
+    local run_test = function (test_script)
+        local final_script = util.render([[
+            local test = require "__test__"
+
+            local function assert_pallene_error(message, target, ...)
+                if [===[$backend]===] == "c" then
+                    local ok, err = pcall(target, ...)
+                    assert(not ok)
+                    assert(string.find(err, message, nil, true))
+                end
+            end
+
+            local function assert_pallene_type(expected, x, ...)
+                if [===[$backend]===] == "c" then
+                    assert(expected == type(x))
+                end
```

Comment:
We could change this function into `assert_is_pallene_record`. On the C backend it tests if the thing is userdata and on the Lua backend it could test if the thing is  a  table.